go/billing/tests/test_api.py

```diff
@@ -259,6 +259,110 @@ def test_low_credit_notification_zero_last_topup_value(self):
         self.assertFalse(mock_task_delay.called)
 
     @inlineCallbacks
+    def test_credit_cutoff_inbound_not_reached(self):
+        self.patch(app_settings, 'ENABLE_LOW_CREDIT_CUTOFF', True)
+
+        mk_message_cost(
+            tag_pool=self.pool1,
+            message_direction=MessageCost.DIRECTION_INBOUND,
+            message_cost=0.2,
+            )
+
+        load_account_credits(self.account, 10)
+
+        transaction1 = yield self.create_api_transaction(
+            account_number=self.account.account_number,
+            message_id='msg-id-1',
+            tag_pool_name='pool1',
+            tag_name='tag1',
+            message_direction=MessageCost.DIRECTION_INBOUND,
+            session_created=False,
+            transaction_type=Transaction.TRANSACTION_TYPE_MESSAGE)
+
+        self.assertFalse(transaction1['credit_cutoff_reached'])
+        self.assertEqual(Transaction.objects.count(), 2)
+
+    @inlineCallbacks
+    def test_credit_cutoff_inbound_reached(self):
+        self.patch(app_settings, 'ENABLE_LOW_CREDIT_CUTOFF', True)
+
+        mk_message_cost(
+            tag_pool=self.pool1,
+            message_direction=MessageCost.DIRECTION_INBOUND,
+            message_cost=0.4,
+            )
+
+        load_account_credits(self.account, 10)
+
+        transaction_kwargs = {
+            'account_number': self.account.account_number,
+            'message_id': 'msg-id-1',
+            'tag_pool_name': 'pool1',
+            'tag_name': 'tag1',
+            'message_direction': MessageCost.DIRECTION_INBOUND,
+            'session_created': False,
+            'transaction_type': Transaction.TRANSACTION_TYPE_MESSAGE
+        }
+
+        yield self.create_api_transaction(**transaction_kwargs)
+        transaction = yield self.create_api_transaction(**transaction_kwargs)
```

Comment:
This was actually something I was wondering about that I didn't ask. The way it currently works is, it requires a message to take the balance below the threshold, and then it blocks all future messages. Should this be changer to block all messages that would take the balance below the threshold?